viewing_party/party.py

```diff
@@ -1,23 +1,197 @@
+# - take one parameter: `user_data`
+#   - the value of `user_data` will be a dictionary with a `"watched"` list of movie dictionaries, and a `"friends"`
+#     - This represents that the user has a list of watched movies and a list of friends
+#     - The value of `"friends"` is a list
+#     - Each item in `"friends"` is a dictionary. This dictionary has a key `"watched"`, which has a list of movie dictionaries.
+#     - Each movie dictionary has a `"title"`.
+# - Consider the movies that the user has watched, and consider the movies that their friends have watched. 
+# Determine which movies at least one of the user's friends have watched, but the user has not watched.
+# - Return a list of dictionaries, that represents a list of movies
+
+def get_unique_watched(user_data):
+    unique_movie = []
+    friends_watched = user_data["friends"]
+    friend_watch_list= []
+    user_watched = user_data["watched"]
+
+    for friends in friends_watched:
+        for movie in friends["watched"]:
+            if movie["title"] not in friend_watch_list:
+                friend_watch_list.append(movie["title"])
+    
+    for user_movie in user_watched:
+        if user_movie["title"] not in friend_watch_list:
+            unique_movie.append(user_movie)
+    print("unique_movies", unique_movie)
+    return unique_movie
+
+def get_friends_unique_watched(user_data):
+    unique_movie = []
+    user_watched_list_titles= []
+    friends_watched_list_title = []
+
+    for user_movie in user_data["watched"]:
+        user_watched_list_titles.append(user_movie["title"])
+    
+    for friends in user_data["friends"]:
+        for movie in friends["watched"]:
+            if movie["title"] not in user_watched_list_titles and movie["title"] not in friends_watched_list_title:
+                friends_watched_list_title.append(movie["title"])
+                unique_movie.append(movie)
+                
+    return unique_movie
 
-        
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
+# 1. Create a function named `get_available_recs`. This function should...
+# - take one parameter: `user_data`
+#   - `user_data` will have a field `"subscriptions"`. The value of `"subscriptions"` is a list of strings
+#     - This represents the names of streaming services that the user has access to
+#     - Each friend in `"friends"` has a watched list. Each movie in the watched list has a `"host"`, which is a string that says what streaming service it's hosted on
+# - Determine a list of recommended movies. A movie should be added to this list if and only if:
+#   - The user has not watched it
+#   - At least one of the user's friends has watched
+#   - The `"host"` of the movie is a service that is in the user's `"subscriptions"`
+# - Return the list of recommended movies
+
+def get_available_recs(user_data):
+    user_not_watched = get_friends_unique_watched(user_data)

```

Comment:
I love seeing you all use your functions you wrote previously inside your code! It lets me know that you are understanding the concepts of writing DRY code!